ios/FluentUI/Vnext/Button/Button.swift

```diff
@@ -52,7 +55,15 @@ struct MSFButtonViewButtonStyle: ButtonStyle {
                                     :
                         AnyView(RoundedRectangle(cornerRadius: tokens.borderRadius)
                                     .fill(Color(isDisabled ? tokens.disabledBackgroundColor :
-                                                    (isPressed ? tokens.highlightedBackgroundColor : tokens.backgroundColor)))))
+                                                    (isPressed ? tokens.highlightedBackgroundColor : tokens.backgroundColor)))
+                                    .shadow(color: isActive ? tokens.pressedShadow1Color : tokens.restShadow1Color,
```

Comment:
if feels little weird to me that we are setting shadowcolor (none color) for regular button as well

Reply:
I don't know if there is a way around that if we're adding the floating button as a style through design tokens. It seems like it would go against the design of design tokens to try and check what style we're making a button for.

Reply:
This is good timing. I'm working on some fixes of animation related problems in the Avatar.
I'll need to add a conditional modifier (#515) for a similar scenario of applying a modifier only if needed instead of creating a neutral value when it does not apply.

@huwilkes you'll be able to use it and remove the none values of the shadow tokens (adding it only to the floating button styles).

Reply:
You can use **.modifyIf()** do something like:

```
AnyView(RoundedRectangle(cornerRadius: tokens.borderRadius)
                                    .fill(Color(isDisabled ? tokens.disabledBackgroundColor :
                                                    (isPressed ? tokens.highlightedBackgroundColor : tokens.backgroundColor)))
                                    .modifyIf(tokens.style == .accentFloating || tokens.style == .subtleFloating, modifications: { view in
                                        view.shadow(color: isActive ? tokens.pressedShadow1Color : tokens.restShadow1Color,
                                                    radius: isActive ? tokens.pressedShadow1Blur : tokens.restShadow1Blur,
                                                    x: isActive ? tokens.pressedShadow1DepthX : tokens.restShadow1DepthX,
                                                    y: isActive ? tokens.pressedShadow1DepthY : tokens.restShadow1DepthY)
                                            .shadow(color: isActive ? tokens.pressedShadow2Color : tokens.restShadow2Color,
                                                    radius: isActive ? tokens.pressedShadow2Blur : tokens.restShadow2Blur,
                                                    x: isActive ? tokens.pressedShadow2DepthX : tokens.restShadow2DepthX,
                                                    y: isActive ? tokens.pressedShadow2DepthY : tokens.restShadow2DepthY)
                                    })
                        ))
```